src/sparc/curation/tools/context_annotations.py

```diff
@@ -11,18 +12,42 @@ def get_dataset_dir():
     return ManifestDataFrame().get_dataset_dir()
 
 
-def get_context_info_file(filename = "scaffold_context_info.json"):
-    dataset_dir = ManifestDataFrame().get_dataset_dir()
-    context_info_dir = dataset_dir
-    if os.path.exists(os.path.join(dataset_dir, "files")):
-        dataset_dir = os.path.join(dataset_dir, "files")
-    if os.path.exists(os.path.join(dataset_dir, "derivative")):
-        context_info_dir = os.path.join(dataset_dir, "derivative")
-    return os.path.join(context_info_dir, filename)
+def update_context_info(context_info):
+    context_info_location = get_absolute_path(get_dataset_dir(), context_info._fileName)
+    metadata_location = get_absolute_path(get_dataset_dir(), context_info._metadata_file)
+    data = context_info.get_context_json()
+    annotation_data = create_annotation_data_json(data)
+    write_context_info(context_info_location, data)
+    update_additional_type(context_info_location)
+    update_supplemental_json(context_info_location, json.dumps(annotation_data))
+    update_derived_from_entity(context_info_location, os.path.basename(context_info._metadata_file))
+    update_parent_source_of_entity(os.path.basename(context_info_location), metadata_location)
 
 
-def get_context_info_dir():
-    return os.path.dirname(get_context_info_file())
+def create_annotation_data_json(data):
+    views = data["views"]
+    samples = data["samples"]
+    annotation_data = {
+        "version": "0.2.0",
+        "id": "sparc.science.annotation_data",
+    }
```

Comment:
Where is the banner entry? (Which is what differentiates this version from the previous one.)